### PR TITLE
net: tcp: Implement Keep-alive support 

### LINKED ISSUE #34044: TCP connection does not support keepalive function
The keepalive function in tcp is very important. When will Zephyr OS support this function?

### PR DESCRIPTION
**Pull request by rlubos**
This PR implements TCP keep-alive support. The PR is based on #46716, however the code has been refactored to address unresolved review feedback and to reduce ifdef slicing of the TCP code.

Additionally, I've added a couple of tests to verify that keep-alive works as expected. I've kept it to a bare minimum though, as testing keep-alive is time consuming (the timeouts are counted in seconds).

Finally, I've decided to create a separte Kconfig page for TCP-related configs. Over time, the number of TCP Kconfig entries increased a lot so it makes sense to group them in a separate page, instead of bloating the generic config space.

FYI @maweicoltsmart. I've kept your signoff on the original commit, to give you credit for the work you've done.

Resolves #34044

### CODE REVIEW

File: subsys/net/lib/sockets/sockets.c [hunk 1]
```diff
@@ -2499,6 +2515,48 @@ int zsock_getsockopt_ctx(struct net_context *ctx, int level, int optname,
...
+							 optval, optlen);
+				if (ret < 0) {
+					errno = -ret;
+					return -1;
+				}
+
+				return 0;
+			}
+
+			break;
```
Comment (jukkar) [inline subsys/net/lib/sockets/sockets.c:2559]: All these options are nearly identical, so could we do it like this:
```
case TCP_KEEPIDLE:
	__fallthrough;
case TCP_KEEPINTVL:
	__fallthrough;
case TCP_KEEPCNT:
			if (IS_ENABLED(CONFIG_NET_TCP_KEEPALIVE)) {
				ret = net_tcp_get_option(ctx, get_tcp_option(optname),
							 optval, optlen);
				if (ret < 0) {
					errno = -ret;
					return -1;
				}

				return 0;
			}
```

and the `get_tcp_option(optname)` could return the correct option according to option name.

File: subsys/net/lib/sockets/sockets.c [hunk 2]
```diff
@@ -3001,6 +3075,48 @@ int zsock_setsockopt_ctx(struct net_context *ctx, int level, int optname,
 			ret = net_tcp_set_option(ctx,
 						 TCP_OPT_NODELAY, optval, optlen);
 			return ret;
+
+		case TCP_KEEPIDLE:
```
Comment (jukkar) [inline subsys/net/lib/sockets/sockets.c:3079]: Same comment to the option setting.

**Review by jukkar — COMMENTED** [review]
Looks good. I noticed one simple optimization possibility in get/set option.

File: subsys/net/lib/sockets/sockets.c (hunk 1 above)
Comment (rlubos) [reply subsys/net/lib/sockets/sockets.c]: Good point - I've optimized option set/get as suggested.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by ssharks — APPROVED** [review]
LGTM

**Pull request merged**